### PR TITLE
Add mypy support

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

**Review by gthvn1 — APPROVED** [review]
Pretty cool :+1: 
First I was afraid... because I thought that also parameters will be annotated.

**Comment by Wescoeur** [comment]
>  First I was afraid... because I thought that also parameters will be annotated.

Only return type is required for a good detection of wrong functions marked as `override`. But I think it's a good idea to use parameter annotation for new modules.

**Comment by stormi** [comment]
Is this going to be submitted upstream? Looks like a large diff on many files, so huge potential for rebase conflicts.

**Comment by Wescoeur** [comment]
> Is this going to be submitted upstream? Looks like a large diff on many files, so huge potential for rebase conflicts.

At least the first 16 commits. I'm waiting for feedback from our side and to have passed the tests before making an upstream PR.

File: drivers/cleanup.py [hunk 1]
```diff
@@ -1200,27 +1206,31 @@ class LVHDVDI(VDI):
     JRN_ZERO = "zero"  # journal entry type for zeroing out end of parent
     DRIVER_NAME_RAW = "aio"
 
-    def load(self, vdiInfo):
+    @override
+    def load(self, info=None) -> None:
+        # `info` is always set. `None` default value is only here to match parent method.
+        assert info
```
Comment (Nambrok) [inline drivers/cleanup.py]: A message for the assert would be good in case it ever happen.

File: tests/test_ISCSISR.py [hunk 2]
```diff
@@ -129,7 +136,7 @@ def setUp(self):
             'tpgt': 'TPGT'
         }
         self.node_records = [(
-            "%s:%d" % (self.node2['ip'], self.node2['port']),
+            f"{self.node2['ip']}:{self.node2['port']}",
```
Comment (Nambrok) [inline tests/test_ISCSISR.py:139]: f-string are only available in python3, would a use of format() make more sense here?

Comment (Wescoeur) [reply tests/test_ISCSISR.py:139]: This PR is only for XCP-ng 8.3. So no issue here. :)

File: drivers/blktap2.py [hunk 3]
```diff
@@ -2244,7 +2283,7 @@ class XenbusDevice(BusDevice):
 
     XBT_NIL = ""
```
Comment (gthvn1) [inline drivers/blktap2.py:2284]: I don't understand why for example XENBUS_DEVTYPE is annotated but XBT_NIL is not?

Comment (Wescoeur) [reply drivers/blktap2.py:2284]: The original def is:
```python
XENBUS_DEVTYPE = None
```

Without the XENBUS_DEVTYPE change we have:
```
> mypy .
drivers/blktap2.py:2427: error: Incompatible types in assignment (expression has type "str", base class "XenbusDevice" defined the type as "None")  [assignment]
Found 1 error in 1 file (checked 127 source files)
```

This issue is fixed using:
```python
XENBUS_DEVTYPE: ClassVar[str] = ""
```

In practice we can directly use:
```python
XENBUS_DEVTYPE = ""
```

But I don't know if's better to explicit the type here. Maybe we can rely on type inference?

Comment (gthvn1) [reply drivers/blktap2.py:2284]: If we start to annotate the class variable we need to do it for all of them IMHO. But if it is too much modification we can rely on type inference for now and we will see. The point is that currently we have some of them that are annotated and others that are not so it is a little bit strange. But also as now I understand the point I will not refuse the PR for that :).

**Review by gthvn1 — APPROVED** [review]
LGTM.

**Comment by gthvn1** [comment]
:tada:

**Pull request merged**